### PR TITLE
Fixes #21

### LINKED ISSUE #21: Immutable array could be modified with some methods
The original `ImmutableArray` object should NOT be changed with `push()`, `unshift()`, `shift()` and `pop` methods. It should return new object with this changes.

Thanks guys from reddit.com [discussion](https://www.reddit.com/r/PHP/comments/3nysjn/the_wrapper_for_all_php_internal_builtin_array/cvsksxm).

### PR DESCRIPTION
**Pull request by znck**
Add `push()` and `unshift()` to `ImmutableArray`

Fixes issue #21

### CODE REVIEW

File: src/ImmutableArray.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ public function reverse($preserveKeys = false)
     /**
      * Pad array to the specified size with a given value
      *
-     * @param int $size Size of result array
+     * @param int   $size  Size of result array
```
Comment (bocharsky-bw) [inline src/ImmutableArray.php]: I think we should avoid this extra spaces.
Anyway, this changes should be in other PR not in this bug fix.

**Comment by znck** [comment]
@bocharsky-bw Removed spaces.

**Comment by bocharsky-bw** [comment]
@znck Nice work! Thanks to fix it!

**Comment by bocharsky-bw** [comment]
I just merged this PR with rebase and squash your commits in https://github.com/bocharsky-bw/Arrayzy/commit/c8382fab8da3bb6d89e37d0a9bf1597acf9c35a2. All your work accepted.

**Pull request closed without merging**